### PR TITLE
feat(eval): make eval simulation seeding reproducible and configurable

### PR DESCRIPTION
**Pull request by shuheng-liu**
## What this does

Reworks how **evaluation** environments are seeded so the eval scene set is deterministic by default and independently controllable. (🗃️ Feature)

- **Rank-consistent by default.** Every accelerator rank now seeds its eval envs identically, so the scene a given `(task, episode)` maps to no longer depends on the world size / node count — an eval on 1 GPU and on 16 GPUs samples the same scenes. Previously each rank *always* diverged via a `process_index * 10000` offset, which made eval results non-reproducible across different GPU counts.
- **`eval.decorrelate_rank_seeds` (opt-in, default `False`).** Restores per-rank scene divergence for the case where a task is *deliberately* replicated across ranks (listed N× to fill N ranks via round-robin sharding) to gain coverage. Instead of the magic `* 10000` (which silently collides once a rank exceeds 10000 episodes), each rank takes an **orthogonal, non-overlapping** slice of the seed line whose stride is the *exact* per-rank span (`n_batches * num_envs`), so the blocks tile with no gap or overlap. It uses the **global** `process_index`, so blocks never collide across nodes.
- **`eval.seed` (default `None`).** Lets you pin the eval *simulations* independently of the top-level `cfg.seed` (e.g. hold the eval scene set fixed while sweeping the training seed). Falls back to `cfg.seed` when unset; the global `set_seed` for model init / dataset shuffling stays on `cfg.seed`.
- **Fix `RoboCasaEnv.reset` double-shift.** It added `episode_index` on top of an already-per-worker-distinct seed. gymnasium's `SyncVectorEnv` / `AsyncVectorEnv` and the eval harness already hand each sub-env a distinct seed, so the extra shift made scene seeds collide across rollout batches (e.g. with `n_envs=4`, batch 0 / slot 2 reused batch 1 / slot 0), yielding fewer distinct scenes than `n_episodes`. Explicit seeds now pass through verbatim; `episode_index` remains the seed only on an *unseeded* reset so workers still differ.

No model, training-loop, optimizer, or sampler behavior changes — the only `train.py` touch is the eval-seed source inside `_eval_with_fresh_envs`.

## How it was tested

CPU unit tests (this is the gate the changes are covered by; no GPU/sim behavior changes):

- Added `TestResetSeed` in `tests/envs/test_robocasa.py`: an explicit seed passes through `RoboCasaEnv.reset` verbatim (no `episode_index` shift), an unseeded reset falls back to `episode_index`, and a per-worker seed range yields contiguous collision-free scene seeds.
- Added `TestRankSeedOffset` in `tests/scripts/test_eval.py`: the default offset is 0 for all ranks (rank-consistent), decorrelated blocks partition `[0, W·span)` exactly (no gap/overlap), and the span-based stride stays collision-free past 10000 episodes/rank.
- Added `TestResolveEvalSeed` in `tests/scripts/test_eval.py`: `eval.seed` takes precedence when set (including the `0` edge case), falls back to `cfg.seed` when `None`, and returns `None` only when both are `None`.
- Updated `TestEvalWithFreshEnvs._cfg` in `tests/scripts/test_train.py` to carry `eval.seed=None`.

Full affected CPU suite green:

```
pytest -m "not gpu and not network" -n auto tests/scripts/test_eval.py tests/scripts/test_train.py tests/configs/ tests/envs/test_robocasa.py
# 251 passed
```

`pre-commit run --all-files`-equivalent on the changed files is clean.

## How to checkout & try? (for the reviewer)

```bash
pytest -sx tests/scripts/test_eval.py::TestRankSeedOffset
pytest -sx tests/scripts/test_eval.py::TestResolveEvalSeed
pytest -sx tests/envs/test_robocasa.py::TestResetSeed
```

Override the new knobs on any eval/training run that uses a sim env:

```bash
# Pin the eval scenes independently of the training seed:
opentau-eval --accelerate-config <yaml> --config_path=<json> --eval.seed=12345
# Opt back into per-rank scene divergence (only when a task is replicated across ranks):
opentau-eval --accelerate-config <yaml> --config_path=<json> --eval.decorrelate_rank_seeds=true
```

## Checklist

- [x] I have added Google-style docstrings to important functions and ensured function parameters are typed.
- [ ] My PR includes policy-related changes.
  - [ ] If the above is checked: I have run the GPU pytests (pytest -m "gpu") and regression tests.

### CODE REVIEW

File: src/opentau/scripts/eval.py [hunk 1]
```diff
@@ -256,6 +256,68 @@ def rollout(
...
+        per_rank_span: The exact number of seeds one rank consumes
+            (``n_batches * num_envs``), used as the (collision-free) block stride.
+
+    Returns:
+        The additive seed offset for this rank: ``0`` when ``decorrelate`` is False,
+        else ``process_index * per_rank_span``.
+    """
+    if not decorrelate:
+        return 0
+    return process_index * per_rank_span
```
[the remark this answers is not in the document]
Comment (shuheng-liu) [reply src/opentau/scripts/eval.py:318]: Agreed this is non-blocking / off-design. There's also a structural reason it can't actually alias redundant scenes: `create_robocasa_envs` builds a rank's envs into a dict keyed by task name (`out[task_name][0] = env_cls(fns)`, `robocasa.py:903`). So if round-robin assigns the same task name to a rank twice (i.e. replication > `world_size`), the second env overwrites the first — a rank only ever holds one env per task name. The surplus replicas are collapsed *before* any seeding, so a `process_index`-keyed offset is the correct granularity (one block per `(task, rank)`).

The supported pattern is replication ≤ `world_size` (one replica per rank, to "fill N ranks"), where every replica lands on a distinct rank and the blocks stay collision-free by construction. Leaving as-is.
